src/main/java/io/codelex/dateandtime/practice/DatePeriod.java

```diff
@@ -0,0 +1,93 @@
+package io.codelex.dateandtime.practice;
+
+import java.time.LocalDate;
+
+public class DatePeriod {
+    private final LocalDate startDate;
+    private final LocalDate endDate;
+
+    public DatePeriod(LocalDate startDate, LocalDate endDate) {
+        this.startDate = startDate;
+        this.endDate = endDate;
+    }
+
+    public DatePeriod intersection(DatePeriod secondDate) {
+        if (startDate.isBefore(secondDate.startDate) &&
+            endDate.isAfter(secondDate.startDate) &&
+            endDate.isBefore(secondDate.endDate)) {
+            return _1overlapStartOf2(secondDate);
```

Comment:
laba ideja padot atbildi ar metodi, bet domāju ka seit tas ir diezgan generic, kur vienmēr atgriezīsi dateperiod ar diviem datumiem, domāju ka šim šeti vari izmantot tikai vienu padodot divus periodus, brīdi kad objekti kļūst apjomīgaki un sarēžģītāki, noteikti laba ideja tā darīt

un varbūt to izmantojot arī sanāktu kādu `else if` izgriezt? not sure 😄

Reply:
Hint: think of all the possible scenarios how two periods can overlap each other before writing code. Write smaller private methods inside DatePeriod class that would help to calculate intersection, ..

Tikai tāpēc izlēmu par papildus metodēm, bet pamēģināju izveidot savādāk, ja metodēs pārbauda un atgriež boolean un tad izvada tikai datuma pārklāšanās periodu.

Neesmu pārliecināta, vai sapratu par -->  ..vari izmantot tikai vienu padodot divus periodus  ?

Vadījos pēc piemēra:
LocalDate secondStart = LocalDate.of(2022, 1, 10);
LocalDate secondEnd = LocalDate.of(2022, 1, 25);
DatePeriod secondPeriod = new DatePeriod(secondStart, secondEnd);

DatePeriod overlap = firstPeriod.intersection(secondPeriod);

Reply:
izdarīji pareizi, par papildus metodēm bija vairāk domāts par if statements pārbaudēm pašām
un par atgriešanu jā, ka var tikai new DatePeriod atgriezt, kam šeit īsti nebija jēgas papildus metodei

Reply:
Paldies!